Cython/Debugger/DebugWriter.py

```diff
@@ -18,6 +18,21 @@
             etree = None
 
 from ..Compiler import Errors
+from ..Compiler.StringEncoding import EncodedString
+
+
+def is_valid_tag(name):
+    """
+    Names like '.0' are used internally for arguments
+    to functions creating generator expressions,
+    however they are not identifiers. 
+
+    See gh-5552
+    """
+    if isinstance(name, EncodedString):
+        if name.startswith(".") and name[1:].isnumeric():
```

Comment:
We just want `.isdecimal()` here, not any numeric character.
```suggestion
        if name.startswith(".") and name[1:].isdecimal():
```